webui/react/src/hooks/useRestApi.ts

```diff
@@ -136,15 +79,14 @@ export const useRestApiSimple =
     const source = axios.CancelToken.source();
 
     dispatch({ type: ActionType.SetLoading, value: true });
-    apiReq({ ...params, cancelToken: source.token })
+    apiRequest({ ...params, cancelToken: source.token })
       .then((result) => dispatch({ type: ActionType.SetData, value: result } ))
       .catch((e) => (!axios.isCancel(e)) && dispatch({ type: ActionType.SetError, value: e }));
 
     return (): void => source.cancel();
-  }, [ apiReq, params ]);
+  }, [ apiRequest, params ]);
 
   return [ state, setParams ];
```

Comment:
non-blocking: as we define it here the second return value just sets the parameter and doesn't request any request any resources directly. IMO we could name them as such eg instead of `requestAgents` we use `triggerAgentsRequest`.